### PR TITLE
extract main view controller segues into extension

### PR DESCRIPTION
**Pull request by brindy**
Task/Issue URL: https://app.asana.com/0/0/1205584556126566/f
Tech Design URL:
CC:

**Description**:

Extracts segues in the main view controller into an extension which is more programatic about opening what used to be storyboard references / segues.

Main.storyboard now only contains MainViewController

SuggestionTray was moved to a new storyboard

All segues were extracted to an extension 

**Steps to test this PR**:

* When testing, please compare with the production app.  There should be no visible difference. 
* Please test iPhone, iPad and all supported versions of iOS

Onboarding
1. Clean install
2. Launch app
3. Run through onboarding

Home Row prompt
1. Edit `showNow()` in `HomeRowReminder` to return true
2. Perform a search
3. Click on the prompt that appears  

Bookmarks
1. Launch bookmarks from the app's Home Screen
2. Browse to a page, launch bookmarks from the menu
3. Bookmark a page and then edit the bookmark from the menu
4. Favorite a page, return to the app's Home Screen and edit the favorite

Dax Action Sheets
1. Clean install of the app
2. Get through onboarding
3. Visit cnn.com -> Dax will tell you that trackers are being blocked, high five!
4. FireButton should be highlighted with pulsing circle
5. Tap fire button -> Dax will tell you about person data build up
6. Clear tabs and data
7. Perform a search -> Dax will tell you your searches are anonymous 
8. Visit Wikipedia -> No Dax dialog

Report Broken Site
1. Visit a website
2. Choose Report Broken Site

Downloads
1. Choose Downloads from the menu

Tab Switcher
1. Tap on the tabs button

Settings
1. Tap on settings from Home Screen 
2. Tap on settings from the menu
3. Turn on autofill in settings
4. Turn on cookie management 
5. Try and login in to Netflix.com and save the account
6. When the toast appears tap view 
7. Visit twitter.com
8. After cookie management kicks in open the privacy dashboard
9. Tap on Cookies Managed
10. Tap on disable in settings

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGoTests/PrivacyIconLogicTests.swift [hunk 1]
```diff
@@ -33,14 +33,6 @@ class PrivacyIconLogicTests: XCTestCase {
     static let ddgMainURL = URL(string: "https://duckduckgo.com")!
     static let ddgSupportURL = URL(string: "https://duckduckgo.com/email/settings/support")!
     
-    override func setUp() {
```
Comment (brindy) [inline DuckDuckGoTests/PrivacyIconLogicTests.swift]: swiftlint on GitHub started complaining about unused overridden function

File: DuckDuckGo/SaveLoginViewController.swift [hunk 2]
```diff
@@ -61,10 +61,6 @@ class SaveLoginViewController: UIViewController {
         viewModel?.viewControllerDidAppear()
     }
     
-    override func viewDidLayoutSubviews() {
```
Comment (brindy) [inline DuckDuckGo/SaveLoginViewController.swift]: SwiftLint override unused function

File: DuckDuckGo/AutofillListItemTableViewCell.swift [hunk 3]
```diff
@@ -118,10 +118,6 @@ class AutofillListItemTableViewCell: UITableViewCell {
             contentStackView.bottomAnchor.constraint(equalTo: margins.bottomAnchor)
         ])
     }
-    
-    override func setSelected(_ selected: Bool, animated: Bool) {
```
Comment (brindy) [inline DuckDuckGo/AutofillListItemTableViewCell.swift]: SwiftLint override unused function

File: DuckDuckGo/KeychainItemsDebugViewController.swift [hunk 4]
```diff
@@ -115,10 +115,6 @@ private enum SecClass: CaseIterable {
 
 class KeychainItemsDebugViewController: UITableViewController {
 
-    override func viewDidLoad() {
```
Comment (brindy) [inline DuckDuckGo/KeychainItemsDebugViewController.swift]: SwiftLint override unused function

File: DuckDuckGo/SettingsViewController.swift [hunk 5]
```diff
@@ -544,10 +529,6 @@ class SettingsViewController: UITableViewController {
         }
     }
     
-    override func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
```
Comment (brindy) [inline DuckDuckGo/SettingsViewController.swift]: SwiftLint override unused function

File: DuckDuckGoTests/TrackerAnimationLogicTests.swift [hunk 6]
```diff
@@ -30,14 +30,6 @@ class TrackerAnimationLogicTests: XCTestCase {
 
     static let pageURL = URL(string: "https://example.com")!
     
-    override func setUp() {
```
Comment (brindy) [inline DuckDuckGoTests/TrackerAnimationLogicTests.swift]: SwiftLint override unused function

File: IntegrationTests/AtbServerTests.swift [hunk 7]
```diff
@@ -37,11 +37,7 @@ class AtbServerTests: XCTestCase {
         loader = StatisticsLoader(statisticsStore: store)
         
     }
-    
-    override func tearDown() {
```
Comment (brindy) [inline IntegrationTests/AtbServerTests.swift]: SwiftLint override unused function

File: DuckDuckGo/MainViewController+Segues.swift [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+        os_log(#function, log: .generalLog, type: .debug)
+
+        let storyboard = UIStoryboard(name: "Bookmarks", bundle: nil)
+        let bookmarks = storyboard.instantiateViewController(identifier: "BookmarksViewController") { coder in
+            BookmarksViewController(coder: coder,
+                                    bookmarksDatabase: self.bookmarksDatabase,
+                                    bookmarksSearch: self.bookmarksCachingSearch,
+                                    syncService: self.syncService,
+                                    syncDataProviders: self.syncDataProviders)
+        }
```
Comment (miasma13) [inline DuckDuckGo/MainViewController+Segues.swift:91]: You missed `bookmarks.delegate = self` here. Without the delegate tapping a bookmark has no effect.

File: DuckDuckGo/MainViewController+Segues.swift [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+
+        let controller = ThemableNavigationController(rootViewController: settings)
+        controller.modalPresentationStyle = .automatic
+        present(controller, animated: true) {
+            completion?(settings)
+        }
+    }
+
+    private func hideAllHilightsIfNeeded() {
```
Comment (miasma13) [inline DuckDuckGo/MainViewController+Segues.swift]: Typo? hilights -> highlights

**Review by miasma13 — CHANGES_REQUESTED** [review]
There are couple of leftover performSegues that trigger assertion in MainViewController:
- AppDelegate.swift L:705
- AppTPHomeViewModel.swift L:90

**Review by miasma13 — APPROVED** [review]
Works as expected 👍

**Pull request merged**